src/ui/components/map/providers/googlemapprovider.js

```diff
@@ -23,6 +23,31 @@ export default class GoogleMapProvider extends MapProvider {
     if (!this.hasValidClientCredentials() && !this._apiKey) {
       throw new Error('GoogleMapsProvider: Missing `apiKey` or {`clientId`, `signature`}');
     }
+
+    /**
+     * Language of the map.
+     * @type {string}
+     */
+    this._language = this.getValidatedLanguage(this._locale);
+  }
+
+  /**
+   * Google Maps supports some langauge codes that are longer than two characters. If the
+   * locale matches one of these edge cases, use it. Otherwise, fallback on the first two
+   * characters of the locale and print a warning.
+   * @param {string} localeStr Unicode locale
+   */
+  getValidatedLanguage (localeStr) {
```

Comment:
What would you think about renaming this to `getLanguage` or `getLanguageForGoogle`? I don't think this method is so much validating a language as just picking the right language for Google, based on the locale.